avl/node.go

```diff
@@ -329,11 +329,13 @@ func (n *node) rehash() {
 
 func (n *node) rehashNoWrite() [MerkleHashSize]byte {
 	buf := bytebufferpool.Get()
-	defer bytebufferpool.Put(buf)
```

Comment:
Not sure if the nanoseconds shaved by not using `defer` is worth sacrificing the convenience it provides though.